src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurDetailLineGroupForCollector.java

```diff
@@ -255,4 +400,50 @@ protected KualiDecimal getSignedAmountFromOriginEntry(OriginEntryFull originEntr
         }
     }
 
+    protected void reportAllLinesForGroupAsFailuresDueToOrphanedCashAdvanceLines(
+            BiConsumer<ConcurStandardAccountingExtractDetailLine, String> lineErrorReporter) {
+        List<ConcurStandardAccountingExtractDetailLine> linesForGroup = getDetailLinesSortedBySequenceNumberNumerically();
+        Set<String> sequenceNumbersForOrphanedCashAdvanceLines = getSequenceNumbersForOrphanedCashAdvanceLines();
+        
+        for (ConcurStandardAccountingExtractDetailLine detailLine : linesForGroup) {
+            String errorMessage = GROUP_WITH_ORPHANED_CASH_ADVANCE_MESSAGE;
+            if (sequenceNumbersForOrphanedCashAdvanceLines.contains(detailLine.getSequenceNumber())) {
+                errorMessage = String.format(ORPHANED_CASH_ADVANCE_MESSAGE_FORMAT,
+                        ORPHANED_CASH_ADVANCE_MESSAGE_FRAGMENT, detailLine.getCashAdvanceKey());
+            }
+            lineErrorReporter.accept(detailLine, errorMessage);
+        }
+    }
+
+    protected Set<String> getSequenceNumbersForOrphanedCashAdvanceLines() {
+        return consolidatedCashAdvanceLines.get(ORPHANED_CASH_ADVANCES_KEY)
+                .stream()
+                .map(ConcurStandardAccountingExtractDetailLine::getSequenceNumber)
+                .collect(Collectors.toCollection(HashSet::new));
+    }
+
+    protected List<ConcurStandardAccountingExtractDetailLine> getDetailLinesSortedBySequenceNumberNumerically() {
+        List<ConcurStandardAccountingExtractDetailLine> linesForGroup = new ArrayList<>();
+        
+        for (ConcurDetailLineSubGroupForCollector subGroup : consolidatedRegularLines.values()) {
+            linesForGroup.addAll(subGroup.getCashLines());
+            linesForGroup.addAll(subGroup.getCorporateCardLines());
+        }
+        for (List<ConcurStandardAccountingExtractDetailLine> cashAdvanceLines : consolidatedCashAdvanceLines.values()) {
+            linesForGroup.addAll(cashAdvanceLines);
+        }
+        
+        Collections.sort(linesForGroup, this::compareBySequenceNumberAsPositiveNumericString);
+        return linesForGroup;
+    }
+
+    protected int compareBySequenceNumberAsPositiveNumericString(
+            ConcurStandardAccountingExtractDetailLine line1, ConcurStandardAccountingExtractDetailLine line2) {
+        int lengthComparison = line1.getSequenceNumber().length() - line2.getSequenceNumber().length();

```

Comment:
I need to compare these Strings as if they were positive integers. With String.compareTo(), the Strings are compared first by character content and then by length; I need it to be the reverse for numeric comparsion. (For example, "10" would precede "2" if I used String.compareTo().) I could alternatively parse them into numbers and then compare them, but that could result in a lot of construct-and-discard operations.